pkg/common/utils/utils.go

```diff
@@ -102,6 +105,22 @@ func GetApplicationIDFromPod(pod *v1.Pod) (string, error) {
 		pod.Spec.String())
 }
 
+func GetNamespaceQuotaFromAnnotation(namespaceObj *v1.Namespace) *si.Resource {
+	log.Logger.Info("### GetNamespaceQuotaFromAnnotation")
+	// retrieve resource quota info from annotations
+	cpuQuota := namespaceObj.Annotations["yunikorn.apache.org/queue.max.cpu"]
+	memQuota := namespaceObj.Annotations["yunikorn.apache.org/queue.max.memory"]
+
+	log.Logger.Info("### GetNamespaceQuotaFromAnnotation", zap.String("cpu", cpuQuota), zap.String("mem", memQuota))
+
+	// no quota found
+	if cpuQuota == "" && memQuota == "" {
+		return nil
+	}
+
+	return common.ParseResource(cpuQuota, memQuota)
```

Comment:
This is the point we need to build in sanity checks with clear logging as to what is checked and failed.:
- zero values for either setting
- negative values for either setting

Only return a resource that has both values set to a value larger than 0 or if only one is specified that one needs to be larger than 0. A nil should be returned in all other cases.

Reply:
After offline discussion, we agree to have such sanity checks in core alone. We do not duplicate them in shim because currently there is no way to send feedback back to the client  event it fails.